kube-runtime/src/wait.rs

```diff
@@ -0,0 +1,114 @@
+use futures::TryStreamExt;
+use kube::{Api, Resource};
+use serde::de::DeserializeOwned;
+use snafu::{futures::TryStreamExt as _, Snafu};
+use std::fmt::Debug;
+
+use crate::watcher::{self, watch_object};
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("failed to probe for whether the condition is fulfilled yet: {}", source))]
+    ProbeFailed {
+        #[snafu(backtrace)]
+        source: watcher::Error,
+    },
+}
+
+/// Watch an object, and Wait for some condition `cond` to return `true`.
+///
+/// `cond` is passed `Some` if the object is found, otherwise `None`.
+///
+/// # Caveats
+///
+/// Keep in mind that the condition is typically fulfilled by an external service, which might not even be available. `await_condition`
+/// does *not* automatically add a timeout. If this is desired, wrap it in [`tokio::time::timeout`].
+///
+/// # Errors
+///
+/// Fails if the type is not known to the Kubernetes API, or if the [`Api`] does not have
+/// permission to `watch` and `list` it.
+///
+/// Does *not* fail if the object is not found.
+pub async fn await_condition<K>(
```

Comment:
I think the naming here is awkward for a public function. We have a module named `wait` and a fn `await_condition` that we have to `.await` leading to `wait::await_condition(cond).await` with full paths.

I think if we let people just import the module `wait` here we could rename this function `until`, and we could have `wait::until(cond).await`, even with the full import `until(cond).await` sounds OK to me.